src/redis.js

```diff
@@ -1,35 +1,67 @@
 const Redis = require('ioredis')
 const LIST_KEY = 'cluster-list'
-const { reduce } = require('fauxdash')
+const CHANNELS_KEY = 'cluster-channels'
 
-function addCluster (client, cluster, url) {
-  return client.hset(LIST_KEY, cluster, url)
+async function addCluster (client, cluster, url, channel) {
+  await client.hset(LIST_KEY, cluster, url)
+  if (channel) {
+    await client.hset(CHANNELS_KEY, cluster, channel)
+    // this enables easy lookup by channel name
+    await client.sadd(`${CHANNELS_KEY}/${channel}`, cluster)
+  }
```

Comment:
As we discussed in pairing, Redis commands can be performed in a transaction to prevent structures from getting out of sync in the event of a failure of a subset of the commands (whether due to a connection issue or a bug).